crates/paimon/src/io/file_io.rs

```diff
@@ -129,10 +148,15 @@ impl FileIO {
     /// Delete a dir recursively.
     ///
     /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-common/src/main/java/org/apache/paimon/fs/FileIO.java#L139>
-    pub async fn delete_dir(&self, path: &str) -> Result<()> {
-        self.op.remove_all(path).await.context(IoUnexpectedSnafu {
-            message: "Failed to delete dir".to_string(),
-        })?;
+    pub async fn delete_dir(&self, path: impl AsRef<str>) -> Result<()> {
+        let (op, relative_path) = self.inner_storage.create_operator(&path)?;
+
+        op.remove_all(relative_path)
+            .await
+            .context(IoUnexpectedSnafu {
+                message: "opendal delete directory failed",
```

Comment:
We don't need to expose the underlying details that `opendal` delete failed.

Reply:
Yes, this should be the same problem as the dependency of the Url library